core/src/test/scala/integration/kafka/api/LogDirFailureTest.scala

```diff
@@ -60,6 +61,37 @@ class LogDirFailureTest extends IntegrationTestHarness {
   }
 
   @Test
+  def brokerWithOldIBPShouldHaltOnLogDirFailure() {
+    @volatile var statusCodeOption: Option[Int] = None
+    Exit.setHaltProcedure { (statusCode, _) =>
+      statusCodeOption = Some(statusCode)
+      throw new IllegalArgumentException
```

Comment:
Do we need to throw an exception here? It seems that we could just check statusCodeOption after the server is started.